### PR TITLE
Fix npm install/link, xdrip cgm sets bt_offline, file tests to preven…

### LINKED ISSUE #1145: oref0-setup.sh fails on npm global-install
This was from a week ago or so. This was also a fresh flash of jubilinux 3.0. Other rig with existing setup was able to install just fine (also jubilinux 3.0). I've tried some of the fixes that folks mention, such as removing package-lock.json, trying the install repeating, adjusting npm proxy, nothing seems to help. @PieterGit may have insight around whether the changes to package.json may be the cause. This is from the `dev` branch
```
Checking oref0 installation
Forcing install of latest oref0 from /home/root/src/oref0/

> oref0@0.7.0-dev global-install /root/src/oref0
> npm install && sudo npm install -g && sudo npm link && sudo npm link oref0

npm WARN deprecated crypto@0.0.3: This package is no longer supported. It's now a built-in Node module. If you've depended on crypto, you should switch to the one that's built-in.
npm notice created a lockfile as package-lock.json. You should commit this file.
added 222 packages from 653 contributors and audited 362 packages in 64.01s
found 0 vulnerabilities

npm ERR! path /usr/lib/node_modules/.staging/oref0-aa44602e/node_modules/abbrev
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/usr/lib/node_modules/.staging/oref0-aa44602e/node_modules/abbrev' -> '/usr/lib/node_modules/.staging/abbrev-fd397e54'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/root/.npm/_logs/2018-11-17T06_06_45_062Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 254
npm ERR! oref0@0.7.0-dev global-install: `npm install && sudo npm install -g && sudo npm link && sudo npm link oref0`
npm ERR! Exit status 254
npm ERR!
npm ERR! Failed at the oref0@0.7.0-dev global-install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/root/.npm/_logs/2018-11-17T06_06_45_237Z-debug.log
No module named mmeowlink.vendors.mmeowlink
Installing latest mmeowlink
```

### PR DESCRIPTION
**Pull request by rkresha**
…t stderr output, add bnep0 interface if doesn't exist, enhance bnep0 cycling logic, cleanup/standardization

Fix npm install/link, xdrip cgm sets bt_offline, file tests to prevent avoidable stderr output, add bnep0 interface if doesn't exist, enhance bnep0 cycling logic, cleanup/standardization of parsing files

Fixes #1145 for me

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "openaps oref0 reference implementation of the reference design",
   "scripts": {
     "test": "make test",
-    "global-install": "npm install && sudo npm install -g && sudo npm link && sudo npm link oref0"
+    "global-install": "npm install && sudo npm link && sudo npm link oref0 && sudo npm install -g"
```
Comment (scottleibrand) [inline package.json]: What does changing this order do?

Comment (rkresha) [reply package.json]: For me it fixed #1145 by allowing global-install to work

**Comment by cascer1** [comment]
I'm testing this now

**Comment by cascer1** [comment]
the first run of `npm run global-install` seems to have worked fine. However, after that I ran `oref0-setup` which took a long time and I eventually ctrl+c'd to get out of it. After that I couldn't run `oref0-setup` again so I did `npm run global-install` a second time. This time it errored out:

image

**Comment by rkresha** [comment]
@cascer1 your version of npm is really old. Mine is v.6.4.1

**Comment by cascer1** [comment]
@rkresha Weird how my npm version is really old though, shouldn't everything be installed by opeanps-setup?

I did a diff between opeanps/oref0:dev and your oref0:dev and don't see any changes. Could you perhaps share them?

**Comment by scottleibrand** [comment]
Is this good to merge?

**Comment by rkresha** [comment]
Good from my standpoint
On 12/9/2018 6:55:57 PM, Scott Leibrand <notifications@github.com> wrote:
Is this good to merge?

**Comment by eupakhomov** [comment]
OpenAps fresh installation on an empty Pi Zero with Explorer HAT with this commit breaks down WiFi connection. No WiFi network is detected on the rig after installation. When reverted to previous commit - works fine.

**Comment by scottleibrand** [comment]
Do you know how exactly this breaks it?

**Comment by eupakhomov** [comment]
@scottleibrand most probably addition of bnep0 interface. I will order second Pi and might play with it to ensure but it will take some time.

**Comment by rkresha** [comment]
Breaks how? Doing a 'sudo ifdown wlan0; sudo ifup wlan0' yields what?

**Comment by cluckj** [comment]
Yeah, I can confirm the bug: dhcpcd on Raspbian is crashing due to the addition to `/etc/network/interfaces`. Removing the lines from `oref0-setup.sh` fixes the issue, while keeping bt_with_wifi functionality. Is adding the bnep0 interface needed for the Edison?

**Comment by eupakhomov** [comment]
@rkresha I had to fork to previous commit as my wife uses the rig actively and so far we have only one - unfortunately I am very limited in contributing useful diagnostic at the moment.

@cluckj @scottleibrand :thumbsup:

**Comment by rkresha** [comment]
Looking at context this particular piece doesn't make use of sudo and if bnep0 is already in there it shouldn't do anything

'printf 'test' >> /etc/network/interfaces
-bash: /etc/network/interfaces: Permission denied'

**Comment by cluckj** [comment]
We could put it under an `is_edison` check?

**Comment by cluckj** [comment]
image

**Comment by rkresha** [comment]
That's odd. I'm running Edison with jubilinux 3 (stretch) and works without issue. Time for a google

**Comment by rkresha** [comment]
@cluckj  if possible can you provide a copy of the /etc/network/interfaces that is currently working? I'm thinking it is how the rpi vs jubilinux goes through the startup process. With jubilinux it copies `/etc/network/interfaces.client` back to `/etc/network/interfaces`, where as rpi perhaps doesn't, leaving the `/etc/network/interfaces` in a odd state since the code that was causing the issue is writing directly to `/etc/network/interfaces`. The only reason I did this, instead of writing to `/etc/network/interfaces.client` is because `oref0-setup.sh` copies `/etc/network/interfaces` to `/etc/network/interfaces.client` and I wasn't placing my code after that point so I'd loose my edits to `/etc/network/interfaces.client`. In hindsight I should've just made the edits to the code after the copy of the file to `/etc/network/interfaces.client` and left `/etc/network/interfaces` alone. After you confirm what your working interfaces file looks like I may add the code back in after the copy of the file like mentioned above.

**Comment by cluckj** [comment]
```
# interfaces(5) file used by ifup(8) and ifdown(8)

# Please note that this file is written to be used with dhcpcd
# For static IP, consult /etc/dhcpcd.conf and 'man dhcpcd.conf'

# Include files from /etc/network/interfaces.d:
source-directory /etc/network/interfaces.d
```

**Comment by rkresha** [comment]
Here is the new PR #1179

**Pull request merged**